dbms/src/Server/TCPHandler.cpp

```diff
@@ -508,6 +509,10 @@ void TCPHandler::sendHello()
     {
         writeStringBinary(DateLUT::instance().getTimeZone(), *out);
     }
+    if (client_revision >= DBMS_MIN_REVISION_WITH_SERVER_DISPLAY_NAME)
+    {
+        writeStringBinary(server.config().getString("display_name", getFQDNOrHostName()), *out);
```

Comment:
It may lead to DNS query per for each connected client.
Better to calculate `getFQDNOrHostName()` just once.